finetuner/callbacks.py

```diff
@@ -0,0 +1,131 @@
+from typing import Optional
+
+
+class BestModelCheckpoint:
```

Comment:
I would use `dataclass` for the templates. It will provide validation features

Reply:
haven't really used dataclasses before..